File: javascript/datadome-examples/interstitial.js
import { Salamoonder } from 'salamoonder-js';

const URL = 'https://example.com/';
const USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/146.0.0.0 Safari/537.36';
const PROXY = 'http://user:pass@ip:port';
const API_KEY = 'sr-YOUR-KEY';

const HEADERS = {
    'User-Agent': USER_AGENT,
    'sec-ch-ua': '"Google Chrome";v="146", "Not-A.Brand";v="8", "Chromium";v="146"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
    'accept-language': 'en-US,en;q=0.9',
};

async function main() {
    const client = new Salamoonder(API_KEY);

    const response = await client.get(URL, {
        headers: HEADERS,
        proxy: PROXY,
        impersonate: 'chrome133a',
    });

    const ddCookie = response.cookies.get('datadome');
    if (!ddCookie) {
        console.error('No DataDome cookie found');
        process.exit(1);
    }

    const interstitialChallenge = await client.datadome.getInterstitialChallenge(
        response.text,
        ddCookie,
        URL,
        { user_agent: USER_AGENT }
    );

    const taskId = await client.task.createTask('DataDomeInterstitialSolver', {
        captcha_url: interstitialChallenge.captcha_url,
        challenge_page: interstitialChallenge.challenge_page,
        user_agent: USER_AGENT,
    });

    const result = await client.task.getTaskResult(taskId);

    if (!result.url) {
        console.error('Failed to solve challenge:', result);
        process.exit(1);
    }

    const cookieResponse = await client.get(result.url, {
        headers: HEADERS,
        proxy: PROXY,
        impersonate: 'chrome133a',
    });
    
    console.log(cookieResponse.text);
}

main();

File: python/datadome-examples/interstitial.py
import asyncio
from salamoonder import Salamoonder
from loguru import logger

# Configuration
URL = "https://example.com/"
USER_AGENT = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/139.0.0.0 Safari/537.36"
PROXY = "http://user:pass@ip:port"
API_KEY = "sr-YOUR-KEY"

HEADERS = {
    "User-Agent": USER_AGENT,
    "sec-ch-ua": '"Google Chrome";v="139", "Not-A.Brand";v="8", "Chromium";v="139"',
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": '"Windows"',
    "accept-language": "en-US,en;q=0.9",
}

async def main():
    async with Salamoonder(API_KEY) as client:
        response = await client.get(URL, headers=HEADERS, proxy=PROXY, impersonate="chrome133a")
        cookies = response.cookies.get('datadome')

        if not cookies:
            logger.error("No DataDome cookie found")
            return

        constructed_url = client.datadome.parse_interstitial_url(response.text, cookies, URL)

        task_id = await client.task.createTask(
            task_type="DataDomeInterstitialSolver",
            captcha_url=constructed_url,
            user_agent=USER_AGENT,
            country_code="us"
        )

        result = await client.task.getTaskResult(task_id)

        if "cookie" not in result:
            logger.error(f"Failed to solve challenge: {result}")
            return

        cookie_str = result["cookie"]
        solved_cookie = cookie_str.split("datadome=", 1)[1].split(";", 1)[0] if "datadome=" in cookie_str else cookie_str.split(";", 1)[0]

        client.session.cookies.set(
            name="datadome",
            value=solved_cookie,
            domain=".example.com",
            path="/",
            secure=True
        )

        response = await client.get(URL, headers=HEADERS, proxy=PROXY, impersonate="chrome133a")

        if response.status_code == 200:
            logger.success(response.text)
            logger.success("Successfully bypassed Interstitial!")
        else:
            logger.error(f"Bypass failed (response: {response.text})")

if __name__ == "__main__":
    asyncio.run(main())